src/webdav4/fsspec.py

```diff
@@ -325,7 +325,10 @@ def pipe_file(self, path: str, value: bytes, **kwargs: Any) -> None:
         self.client.upload_fileobj(buff, path, **kwargs)
 
     def put_file(
-        self, lpath: "PathLike[AnyStr]", rpath: str, **kwargs: Any
+        self,
+        lpath: "PathLike[AnyStr]",
+        rpath: str,
+        **kwargs: Any,
```

Comment:
Does the `callback` in fsspec expect a callable too? I thought it has to be inherited from [`Callback`]().
If yes, then we may be able to only change this function, with something like following in the else condition.

```python
progress_callback: Optional["CallbackFn"] = None
if callback is not None:
    callback.set_size(os.path.getsize(lpath))
    progress_callback = cast("CallbackFn", callback.relative_update)
```

Type casting is fine in this case. WDYT?

Reply:
@isidentical, if you are worried about `getsize` calls, we could implement `upload_fobj` within the `WebdavFileSystem`, we need that anyway for the `_upload_fobj` in DVC anyway. We could extend it with the support for the callback.